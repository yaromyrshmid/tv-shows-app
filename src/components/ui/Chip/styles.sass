@import "~theme"
    
.chip
    background-color: $primary
    border-radius: 12px
    height: 22px
    margin: 0
    padding: 4px 16px 2px
    text-align: center
    color: $textWhite

.chip-secondary
    background-color: $secondary