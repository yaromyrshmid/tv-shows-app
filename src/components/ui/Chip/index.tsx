import React from "react";

import "./styles.sass";

interface ChipProps {
  children: string;
  color?: "primary" | "secondary";
}

const Chip: React.FC<ChipProps> = ({
  children,
  color = "primary",
}): JSX.Element => <span className={`chip chip-${color}`}>{children}</span>;

export default Chip;
