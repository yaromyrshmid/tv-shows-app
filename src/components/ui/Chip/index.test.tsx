import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Chip from "./index";

describe("Chip component", () => {
  it("Renders", () => {
    render(<Chip>test</Chip>);

    const chipElement = screen.getByText(/test/i);
    expect(chipElement).toBeInTheDocument();
  });

  it("Matches snapshot", () => {
    const component = renderer.create(<Chip>test</Chip>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Contains passed text", () => {
    const chip = shallow(<Chip>test</Chip>);

    expect(chip.text()).toEqual("test");
  });

  it("Has chip-color class when color is passed", () => {
    const chip = shallow(<Chip color="secondary">test</Chip>);

    expect(chip.hasClass("chip-secondary")).toEqual(true);
  });

  it("Has chip-primary class when no color is passed", () => {
    const chip = shallow(<Chip>test</Chip>);

    expect(chip.hasClass("chip-primary")).toEqual(true);
  });
});
