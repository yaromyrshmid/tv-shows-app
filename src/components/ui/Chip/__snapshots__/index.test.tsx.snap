// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip component Matches snapshot 1`] = `
<span
  className="chip chip-primary"
>
  test
</span>
`;
